forui/lib/src/widgets/slider/slider.dart

```diff
@@ -0,0 +1,342 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+import 'package:forui/forui.dart';
+import 'package:forui/src/widgets/slider/layout.dart';
+import 'package:meta/meta.dart';
+
+import 'package:forui/src/widgets/slider/inherited_data.dart';
+
+/// An input where the user selects a value from within a given range.
+///
+/// See:
+/// * https://forui.dev/docs/slider for working examples.
+/// * [FContinuousSliderController.new] for selecting a single continuous value.
+/// * [FContinuousSliderController.range] for selecting continuous range.
+/// * [FDiscreteSliderController.new] for selecting a discrete value.
+/// * [FDiscreteSliderController.range] for selecting a discrete range.
+/// * [FSliderStyles] for customizing a slider's appearance.
+class FSlider extends StatefulWidget {
```

Comment:
Might make sense to make this a `FormField` since sliders are probably part of a form input.

Reply:
I'm open to it, but I'm unsure how/where to position the description and error messages, or if we should even bake those in.